pkg/feature/builder.go

```diff
@@ -13,58 +13,48 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/client/config"
 
-	v1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
 	featurev1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/features/v1"
 	infrav1 "github.com/opendatahub-io/opendatahub-operator/v2/infrastructure/v1"
 )
 
 type partialBuilder func(f *Feature) error
 
 type featureBuilder struct {
-	name     string
-	config   *rest.Config
-	builders []partialBuilder
-	fsys     fs.FS
+	name            string
+	config          *rest.Config
+	builders        []partialBuilder
+	featuresHanlder *FeaturesHandler
+	fsys            fs.FS
 }
 
-func CreateFeature(name string) *featureName { //nolint:golint,revive //No need to export featureBuilder.
-	return &featureName{
+func CreateFeature(name string) *usingFeaturesHandler { //nolint:golint,revive //No need to export featureBuilder.
+	return &usingFeaturesHandler{
 		name: name,
 	}
 }
 
-type featureName struct {
+type usingFeaturesHandler struct {
 	name string
 }
 
-func (fn *featureName) With(spec *v1.DSCInitializationSpec) *featureSource {
-	return &featureSource{
-		spec: spec,
-		name: fn.name,
-	}
-}
-
-type featureSource struct {
-	spec *v1.DSCInitializationSpec
-	name string
-}
-
-func (fo *featureSource) From(source featurev1.Source) *featureBuilder {
+func (u *usingFeaturesHandler) For(featuresHandler *FeaturesHandler) *featureBuilder {
```

Comment:
`For` is reborn!

Reply:
`OwnedBy`, `HandledBy`... naming is hard. The idea here is that handler will have an ownership and control over this feature. Somewhat tightly coupled using the builder, but OTOH this makes it easier to... handle... on the consuming side (controllers)